### PR TITLE
Lesson 3: Homework - Duong Van Truong

### PR DESCRIPTION
**Pull request by tadashii1417**

### CODE REVIEW

File: homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * @return {number}
+ */
+
+function swap(arr, i, j){
+  [arr[i], arr[j]] = [arr[j], arr[i]];
+}
+
+function partition(nums, left, right) {
+  boundary = left;
+  pivot = nums[right];
```
Comment (trongtai37) [inline homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js]: I recommend you using `let` or `const` to declare variables.
This code might throw errors when you run javascript with strict mode.

File: homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const pivotIndex = partition(nums, left, right);
+
+  if (pivotIndex === pos) {
+    return nums[pos];
+  } else if (pivotIndex > pos) {
+    return findKthLargestUtil(nums, pos, left, boundary -1);
+  } else {
+    return findKthLargestUtil(nums, pos, boundary + 1, right);
+  }
+}
```
Comment (trongtai37) [inline homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js]: Just a suggestion, for me if there is `return` statement inside `if` block, i will consider not using `else` or `else if`. This block code can be more concise as below:
```suggestion
  if (pivotIndex === pos) {
    return nums[pos];
  } 
  
  if (pivotIndex > pos) {
    return findKthLargestUtil(nums, pos, left, boundary -1);
  }
  
   return findKthLargestUtil(nums, pos, boundary + 1, right);
}
```

File: homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js (hunk 1 above)
Comment (ted-mfv) [reply homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js]: Thank you for comment. 
I am now coding with both `python` and `javascript`, that's why sometime I use the wrong syntax 👍

File: homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js (hunk 2 above)
Comment (ted-mfv) [reply homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js]: I liked your approach, really appreciate your comment 🙇

File: homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js (hunk 1 above)
Comment (trongtai37) [reply homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js]: Yes, I see, i recommend you using [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) and [ESlint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) extension on VSCode, they can throw these issues in writing code context.

File: homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js (hunk 2 above)
Comment (trongtai37) [reply homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js]: Glad to hear it helps you!

**Pull request merged**